android/buildinf.h

```diff
@@ -0,0 +1,143 @@
+#if !__ANDROID__
+    #error "This header file should only be used for building Android native library."
+#endif
+
+#ifdef __ARM_ARCH
+    #if __ARM_ARCH_ISA_A64
```

Comment:
these don't look like compiler-provided defines, where are they from?
please use the ones both GCC and Clang provide natively (e.g. __arm__, __aarch64__)

Reply:
These are the ones provided by the compiler. I've got those by doing "cc -E -dM" on an empty header file.

Reply:
http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.dui0774a/chr1383660321827.html

Reply:
ok, though I still don't see `__ARM_ARCH_ISA_ARM` in the page 🤔

Reply:
Yeah, I only found it in this doc: https://static.docs.arm.com/ihi0053/c/IHI0053C_acle_2_0.pdf
I don't insist on having `ISA_*` TBH, we can use  `__arm__` and `__aarch_64__`.

Reply:
it's ok, I just picked up some old ndk's gcc and didn't see them there, but even old versions of android clang seem to have it